### ISSUE TITLE
There is no benchmark Compared to tokio and async_std?

### ISSUE THREAD

**Issue by zhuxiujia** [smol-rs/smol#210]
There is no benchmark Compared to tokio and async_std?

**Comment by notgull** [comment]
Benchmarking async runtimes is really hard. The potential workloads are so diverse that, at a point, it doesn't really matter.

As a general rule, there are two main bottlenecks in an `async` runtime: the reactor and the executor. `async-std` and `smol` use the same executor. `tokio`'s executor is probably faster for longer term workloads (like if a network connection slowly dispatches packets of bytes over a long period of time), while `smol`'s is better for shorter term workloads (like all the bytes being immediately available from a socket).

`smol`'s reference executor is almost certainly slower than `tokio`'s executor. But it's easy to slot another executor in (like `smolscale`).

These are just qualitative estimates based on the architectures of the runtimes. If the tokio team would like to collaborate on a more quantitative set of benchmarks I would be happy to oblige.

**Comment by varbhat** [comment]
![IMG_20240201_131427_828](https://github.com/smol-rs/smol/assets/68165364/16f37713-752c-4665-ad16-9ca06b294a87)


I benchmarked axum powered by tokio vs axum powered by smol.

The top section of the image is benchmark of axum powered by tokio. The bottom section of the image is benchmark of axum powered by smol.

The axum endpoint just returns static string "Hello world". It's just homepage example of axum.

CC: @notgull 

TLDR:
axum powered by tokio =>
Reqs/sec : 79.1k
Transfer/sec: 98.1MB

axum powered by smol =>
Reqs/sec: 67.5k
Transfer/sec: 83.6MB

**Comment by zhuxiujia** [comment]
> ![IMG_20240201_131427_828](https://private-user-images.githubusercontent.com/68165364/301424481-16f37713-752c-4665-ad16-9ca06b294a87.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4NDE2NTQsIm5iZiI6MTcwNjg0MTM1NCwicGF0aCI6Ii82ODE2NTM2NC8zMDE0MjQ0ODEtMTZmMzc3MTMtNzUyYy00NjY1LWFkMTYtOWNhMDZiMjk0YTg3LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDAyMzU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5NTZiZGU1NmFiNTUwYTM2ZGJkYTMxN2JkMzA2ZmRkYmU3YWZlNzllMThkNGQ2NWM3MzBhZTU3MTUxNjY2ZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.C0nHhzvY4AJUVZCjJi_5jWDjVIejrmDtqlCUBWjnRKw)
> 
> I benchmarked axum powered by tokio vs axum powered by smol. wrk tool was used to benchmark both.
> 
> The top section of the image is benchmark of axum powered by tokio. The bottom section of the image is benchmark of axum powered by smol.
> 
> The axum endpoint just returns static string "Hello world". It's just homepage example of axum.
> 
> CC: @notgull
> 
> TLDR: axum powered by tokio => Reqs/sec : 79.1k Transfer/sec: 98.1MB
> 
> axum powered by smol => Reqs/sec: 67.5k Transfer/sec: 83.6MB

This doesn't seem as good as Tokio

**Comment by notgull** [comment]
The default executor for `smol` is quite slow. [`smolscale`](https://docs.rs/smolscale/latest/smolscale/) probably has better performance characteristics.

**Comment by sirius-ccwu** [comment]
https://zenoh.io/blog/2022-04-14-rust-async-eval/
